lighthouse/service-selection/multi-cluster-service-selection.md

```diff
@@ -0,0 +1,513 @@
+   2. Requests count - Collected by the Submariner DNS Plugin - Lighthouse and published to Promethues
+   3. Egress cost - different cloud providers have different Networking cost, some charge for traffic within the same provider, some charges
+   for leacing a certain zone, etc. We are going to use the open APIs of the different providers fetch the pricing tables and use them.
+   - See - [Submariner monitoring](https://submariner.io/operations/monitoring/) for full list of the data points available.
+3. Optimization module - the optimization module will be
+   part of the Broker cluster control plan, it will gather the
+   metrics from the different metrics modules calculate the
+   weights for each service and update them in the different clusters.
+   - The optimization module is an external system to Submariner
+
+- If time will pressure, or we find it better we might
+  combine the Metrics and Optimization modules and
+incorporate them withing the Lighthouse for local simple
+and easy solution.
+
+#### Optimized
+
+1. Load-Balancing module - repurpose the module from the first stage
+1. Metrics module - repurpose the module from the first stage
+1. Optimization module - similarly to the first stage, the module
+   will be part of the
+   Broker, just now it will solve the optimization problem described
+   - The optimization module is an external system to Submariner
+1. Estimator Module - will be in-charge of estimating capacity and load if needed
+   - The Estimator module module is an external system to Submariner
+
+### Test plan
+
+Besides implementing Unitests and E2E tests which will be
+configured according to the load balancing algorithm
+during the actual implementation process. We plan on
+experimenting with as close to possible real-life
+scenarios.
+
+#### Parameters
+
+1. Optimization Objective Variables
+    1. RTT -> Latency:
+       latency between Data Centers is usually
+       [steady](http://kth.diva-portal.org/smash/get/diva2:1334280/FULLTEXT01.pdf).
+       Unpredicted events both in the network layer (packet loss,
+       rate-limiting, delays due to congestion, and more), as well
+       as application layer and system-related events (killing
+       /replacing containers, disconnecting cluster and nodes,
+       application bugs more) can cause latency and connectivity
+       issues. There are numerous models and
+       distributions for generating such disruptions.
+        1. Testing Tool - We plan to use [Pumba](https://github.com/alexei-led/pumba)
+           which is a chaos testing tool for Docker containers. It allows
+           performing network emulation and application disruptions.
+    1. Pricing:
+       One of the key objectives in this EP is to
+       improve the price one pays for service to service
+       communication. While measuring the actual pricing is a
+       complicated function to build, of different variables such

```

Comment:
We can make use of topology keys as well to determine the pricing.

]https://kubernetes.io/docs/concepts/services-networking/service-topology/

Reply:
Yea, this is a great idea! I will add it to the proposal!